build_runner/bin/build_runner.dart

```diff
@@ -31,7 +30,8 @@ Future<Null> main(List<String> args) async {
   if (commandName != _generateCommand) {
     logListener = Logger.root.onRecord.listen(stdIOLogListener);
   }
-  var buildScript = await generateBuildScript();
+  var configKey = parsedArgs.command['config'] as String;
```

Comment:
I don't really like that this is assuming the `config` option exists on all commands (although I don't see why that would change either). Maybe just check that `config` is a valid option first for future compatibility? It would be pretty unlikely that we would catch it with a test if we did add another command that didn't support this option.

Reply:
I'm almost tempted to revert this part of it and not allow the `--config` flag to impact build script generation. Effectively it would mean that only `targets` can be changed, not `builders`. Thoughts? It could end up making sense if we decide to split targets and builders into different files. #885

Reply:
Sounds good to me as long as we add in some error handling to validate you don't have a builders section.

Reply:
Updated #885 to mention that we should validate this case. I can't think of a simple way do to it in this PR without expanding the scope a lot (or needing a hack like this arg) so I'll go with ignoring it for now.